src/util/resourceDetails.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { extractResourceDetailsFromUrl } from "./resourceDetails";
+
+describe("extractResourceDetailsFromUrl", () => {
+  it("extracts resource details from lxd entity urls", () => {
+    const serverPath = "/1.0";
+    const instancePath = "/1.0/instances/instance-1?project=default";
+    const storageVolumePath =
+      "/1.0/storage-pools/default/volumes/virtual-machine/volume-test?project=default&target=node-1";
+    const instanceSnapshotPath = "/1.0/instances/instance-1/snapshots/snap0";
+    const volumeSnapshotPath =
+      "/1.0/storage-pools/default/volumes/custom/volume-test/snapshots/snap0";
```

Comment:
shall we split this into one test case for each url?

Reply:
sounds good :+1: